### PR TITLE
chore: added dependabot for npm

### PR DESCRIPTION
**Pull request by ovflowd**
This PR enables `dependabot` with the grouping of dependencies. (https://github.blog/changelog/2023-06-30-grouped-version-updates-for-dependabot-public-beta/)

### CODE REVIEW

**Review by bmuenzenmeyer — APPROVED** [review]

File: .github/dependabot.yml [hunk 1]
```diff
@@ -3,5 +3,40 @@ updates:
...
+          - "@vcarl/remark-headings"
+          - "remark-gfm"
+          - "@mdx-js/*"
+      react:
+        pattersn:
+          - "react"
+          - "react-dom"
+          - "@types/react"
+          - "@types/react-dom"
     open-pull-requests-limit: 10
```
Comment (mikeesto) [inline .github/dependabot.yml:42]:
```suggestion
```
Assuming we don't need this twice?

File: .github/dependabot.yml [hunk 2]
```diff
@@ -3,5 +3,40 @@ updates:
...
+          - "next-sitemap"
+          - "next-themes"
+          - "@vercel/*"
+      remark-mdx:
+        patterns:
+          - "@vcarl/remark-headings"
+          - "remark-gfm"
+          - "@mdx-js/*"
+      react:
+        pattersn:
```
Comment (mikeesto) [inline .github/dependabot.yml]:
```suggestion
        patterns:
```

**Review by mikeesto — APPROVED** [review]

**Comment by mikeesto** [comment]
Appears prettier is not happy

File: .github/dependabot.yml (hunk 1 above)
Comment (bmuenzenmeyer) [reply .github/dependabot.yml:42]: I think the key is declared once for each `package-ecosystem` - so this looks correct to me

**Review by bmuenzenmeyer — COMMENTED** [review]
https://github.com/nodejs/nodejs.org/pull/5493/commits/1aaaad5701d003032081d7d2ce3ab6aa4140110e this seems like the sorta thing lint-staged should have caught. investigating

**Comment by ovflowd** [comment]
> https://github.com/nodejs/nodejs.org/commit/1aaaad5701d003032081d7d2ce3ab6aa4140110e this seems like the sorta thing lint-staged should have caught. investigating

Not really, as I've made the change directly from GitHub's UI. Not locally.

image

**Comment by bmuenzenmeyer** [comment]
> Not really, as I've made the change directly from GitHub's UI. Not locally.

ah, i am glad actually, that explains it

**Pull request merged**